Source/Spring2022_Capstone/Weapon/WeaponBase.cpp

```diff
@@ -86,7 +86,10 @@ void AWeaponBase::Overheat()
 	GEngine->AddOnScreenDebugMessage(-1, 2.f, FColor::Red, TEXT("OVERHEATING"));
 	bIsOverheating = true;
 	bCanFire = false;
-	
+	if (OnWeaponOverheatChangedDelegate.IsBound())
```

Comment:
There's no need to check if it's bound. Not sure if it's the same Execute() call but check on documentation to fix this.
Yes, I made the same mistake before and have kept an eye on it to fix it.